org.eclipse.xtext/src/org/eclipse/xtext/resource/containers/StateBasedContainer.java

```diff
@@ -43,20 +41,8 @@ public StateBasedContainer(IResourceDescriptions descriptions, IContainerState s
 	}
 	
 	@Override
-	protected Iterable<IEObjectDescription> filterByURI(Iterable<IEObjectDescription> unfiltered) {
-		return Iterables.filter(unfiltered, new Predicate<IEObjectDescription>() {
-			private Collection<URI> contents = null;
-
-			@Override
-			public boolean apply(IEObjectDescription input) {
-				if(contents == null) {
-					contents = state.getContents();
-				}
-				URI resourceURI = input.getEObjectURI().trimFragment();
-				final boolean contains = contents.contains(resourceURI);
-				return contains;
-			}
-		});
+	protected Iterable<IResourceDescription> filterByURI(IResourceDescriptions unfiltered) {
```

Comment:
Therefore I'd think an overridden `getExportedObjectsByType` is more straight forward instead of the template method `filterByURI`

Reply:
i was templating it so other implementers of ResourceDescritpionsBasedContainer can make use of that performance improvement too.